src/utils/task/task.go

```diff
@@ -407,5 +407,13 @@ func (self *Task) StopWait() {
 	}
 
 	self.Log.Info("Stopped")
+}
 
+func (self *Task) WithStopChannel(channel chan interface{}) *Task {
+	self.WithSubtaskFunc(func() error {
+		<-channel
```

Comment:
Wait on Ctx.Done() as well. 
Call self.Stop() only if channel was triggered, not the context.

Reply:
done